### ISSUE TITLE
Asynchronous Loop is Failing while using async method

### ISSUE THREAD

**Issue by SugumarVenkatesan** [dask/s3fs#503]
while using  the asynchronous=True in the s3fs constructor, getting running out of loop when using async method. the same is not happening with sync method.

The below snippet is working fine with "_info" sync method

```python
async def gets3fileinfo():
    fs = s3fs.S3FileSystem(key="", secret="", asynchronous=True)
    session = await fs.set_session()
    bucket="xxxxxx"
    key="xxxxxxx"
    input_file = os.path.join(bucket,key)
    print(input_file)
    fsData = await fs._info(input_file)
    pprint(fsData)
    await session.close()
asyncio.run(gets3fileinfo())
```

The same snippet is failing while using "info" async method

```python
async def gets3fileinfo():
    fs = s3fs.S3FileSystem(key="", secret="", asynchronous=True,loop=asyncio.get_running_loop())
    session = await fs.set_session()
    bucket="xxxxxx"
    key="xxxxxxx"
    input_file = os.path.join(bucket,key)
    print(input_file)
    fsData = await fs.info(input_file)
    pprint(fsData)
    await session.close()

asyncio.run(gets3fileinfo())
```

Getting asyncio "Loop is not running issue"

- Dask version: '2021.07.0'
- Python version: 3.8.8
- Operating System: MacOS
- Install method (conda, pip, source): Pip

**Comment by martindurant** [comment]
TL;DR: in async mode, only use the async `_*` methods, not any of the methods that call `sync()` internally

--

We did briefly experiment with completely unifying the APIs, but found that having sync and async methods separate led to much easier to maintain code and fewer subtle errors where calling code didn't know which type of runtime it would be in.

Note that you do not need to pass `loop=` to the constructor if it's being called from within a coroutine anyway. It is possible to make the instance outside of the coroutine and pass it around, but call it with `await` within; and in this case you should give the same loop that will run the coroutine. Sync method may even work in this case (when called from sync code), but I would say that an async instance should only be called from a coroutine by awaiting async methods.

**Comment by SugumarVenkatesan** [comment]
Thanks for the details. It helped.
